### DISCUSSION TITLE
No kernel in my local deployed which serve by *jlpm serve*

### DISCUSSION THREAD

**Discussion by JackYinpei** [jupyterlite/jupyterlite#1208] [Q&A]
I want to create an issue on GitHub: "I tried to deploy jupyterlite on my own ubuntu server according to the method introduced in this article https://jupyterlite.readthedocs.io/en/latest/contributing.html

1. Environment information
    System version: wsl2 Ubuntu 22.04.2
    nodejs v20
    jupyterlab v4.0.7
    python 3.10
2. The dependencies were installed using python -m pip install -r requirements-docs.txt -r requirements-lint.txt instead of using `mamba`
3. `doit` was not used to start the service. After installing node dependencies with `jlpm`, `build` was run followed by starting the service with `jlpm serve` listening on port 5000
4. I would like to know:
      How to introduce the `pyodide` kernel in my local service. I have read many articles online but could not find a solution.
5. I would be very grateful for any suggestions or solutions you may have."
![image](https://github.com/jupyterlite/jupyterlite/assets/49555788/0434de4e-d801-46a0-8597-703447b11119)

**Comment by jtpio** [comment]
> How to introduce the `pyodide` kernel in my local service. I have read many articles online but could not find a solution.

You would need to install the `jupyterlite-pyodide-kernel` package before running `jupyter lite build`.

> I tried to deploy jupyterlite on my own ubuntu server according to the method introduced in this article https://jupyterlite.readthedocs.io/en/latest/contributing.html

There is a dedicated guide to deploy to a standalone server: https://jupyterlite.readthedocs.io/en/stable/quickstart/standalone.html

**Reply by JackYinpei** [reply]
Thank you very much for your prompt reply.

However, as I mentioned in my issue, I started the service using `jlpm` instead of `jupyter lite serve`.
 
I may not have a thorough understanding of this project. I think starting the service with jlpm is different from starting with jupyter lite serve. The reason I used the clone, build, serve approach is that I want to modify the source code of this project. I don't think installing via pip install jupyter lite allows me to modify the source code to meet my needs. 

by the way (I was able to deploy jupyterlite on my computer using jupyter lite init, pyodide was also correctly applied, and the project ran normally)

I also just tried installing `jupyterlite-pyodide-kernel`.
![image](https://github.com/jupyterlite/jupyterlite/assets/49555788/269d4259-f9ed-45bc-ba2f-b4368c1d3323)
Then I executed `jlpm build`  and `jlpm serve ` again, but there is still no pyodide kernel on my website.

I'm not sure if I'm asking some silly questions here, if so I apologize. Also, thank you again for taking the time to respond to my question.

**Reply by jtpio** [reply]
> I think starting the service with jlpm is different from starting with jupyter lite serve

`jlpm` is a pinned version of yarn 3 bundled with JupyterLab, which is used for convenience here for installing JavaScript packages. So it is not meant to be used to start a server. As an alternative to `jupyter lite serve` you can try `python -m http.server`.

> I want to modify the source code of this project

Curious about what you would like to modify? The JupyterLite / JupyterLab ecosystem can be customized to a great extent via third-party extensions that can be developed separately without having to modify the source code. With extensions you can provide new features or change the way some of the core components work.

**Reply by JackYinpei** [reply]
Our students are young and I don't want to give them too many features. I hope to remove some of the advanced functions to prevent students from operating incorrectly.

**Reply by JackYinpei** [reply]
Also, thank you for this project and your contributions to open source. Your project will be of great help to our students.
❤️❤️❤️❤️

**Reply by jtpio** [reply]
> I hope to remove some of the advanced functions to prevent students from operating incorrectly.

Sure. Maybe you can then look into setting a list of `disabledExtensions` to disable some of the core plugins and extensions: https://jupyterlite.readthedocs.io/en/stable/howto/configure/advanced/extensions.html#disabling-extensions-at-runtime
